content/blog/esp32-memory-map-101/index.md

```diff
@@ -0,0 +1,351 @@
+---
+title: "ESP32's family Memory Map 101"
+date: 2024-08-20T18:56:49+02:00
+tags: ["ESP32", "ESP32-S2", "ESP32-S3", "ESP32-C3", "ESP32-C6"]
+showAuthor: false
+authors:
+  - "marek-matej"
+---
+
+
+The performance and memory capabilities of the microcontrollers are increasing, and so is their complexity, especially when the users are responsible for mapping external memory chips such as Flash or SPIRAM by configuring a memory management unit.
+
+When I started to use Espressif ESP32’s series SoC’s with the [Zephyr RTOS](https://docs.zephyrproject.org/latest/index.html), quickly I realized that these SoC’s are a very different world, compared to my previous MCU experiences, which were mostly ARM-related. ARM Cortex-M’s - in most cases - uses Von-Neumann memory architecture, while the ESP32’s series of microcontrollers uses the Harvard memory architecture, which adds complexity.
+
+The primary source of knowledge about any SoC was always the datasheet or Technical Reference Manual of the part. Indeed it still is, but I found the ESP32's TRM — although very useful — quite challenging to read. Especially because there is a significant portion of important information buried in the long textual descriptions.
+
+This sparked the motivation to compile a visually rich and informative depiction of crucial information from the current versions of a TRM. This is intended to be used together with the SoC’s datasheets and TRM to help with a better understanding of the ESP32's family SoCs internals, which is crucial during the system and driver development - for example when porting to other platforms such as Zephyr RTOS.
+
+## ESP32
+
+The ESP32 is the Xtensa LX6 dual-core microcontroller, and it is the oldest SoC from the ESP32’s family of MCUs.
+
+The SoC has access to external Flash memory using the SPI interface. It can also access the external SPIRAM, but only on the data bus.
+
+Zephyr-RTOS support for the ESP32 devkit board can be found [here](https://docs.zephyrproject.org/latest/boards/espressif/esp32_devkitc_wrover/doc/index.html).
+
+
+**ROM**
+
+ROM memories hold the primary bootloader code and other library functions available to the end users. The content of the ROM memories is baked during the manufacturing process and cannot be changed. Please refer to the ECO versions of the ESP32 microcontroller for more detailed information. TLDR is to use the last ECOv3 and later ESP32 chip versions.
```

Comment:
ECO is a term referring to an internal document describing the changes. Lately, there was an initiative to scrape it off release notes in some projects.

The suggested replacement is _chip revision_. The chip revisions are documented in the public document called [Errata](https://www.espressif.com/sites/default/files/documentation/esp32_errata_en.pdf#page=6).

Reply:
Fixed